packages/cli/src/metadataGeneration/metadataGenerator.ts

```diff
@@ -214,17 +214,34 @@ export class MetadataGenerator {
 
   public AddReferenceType(referenceType: Tsoa.ReferenceType) {
     if (!referenceType.refName) {
-      return;
+      throw new Error('no reference type name found');
     }
-    this.referenceTypeMap[decodeURIComponent(referenceType.refName)] = referenceType;
+    this.referenceTypeMap[referenceType.refName] = referenceType;
```

Comment:
decodeURIComponent() not interesting here, because typeResolver guarantees, that only OpenApi-accepted names can be generated.